### PR TITLE
RAC-576: Introduce MF read model & public API for onboarder

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

- New read model dedicated to onboarder representing the measurement family (lives in PHP publicAPI)
- New query dedicated to onboarder to fetch all measurement families in the form of the onboarder read model

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [x] Migration & Installer
- [x] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [x] Tech Doc

### CODE REVIEW

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/FindAllMeasurementFamilies.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Tool\Bundle\MeasureBundle\PublicApi\Onboarder;
+
+use \Akeneo\Tool\Bundle\MeasureBundle\Model\MeasurementFamily as MeasurementFamilyAggregate;
```
Comment (ValentinMerlet) [inline src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/FindAllMeasurementFamilies.php:7]: Is there any reason for the `\` at the beginning of the line? :thinking:

File: src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/FindAllMeasurementFamilies.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        $this->measurementFamilyRepository = $measurementFamilyRepository;
+    }
+
+    /**
+     * @return MeasurementFamily[]
+     */
+    public function execute(): array
+    {
+        $allMeasurementFamilies = $this->measurementFamilyRepository->all();
```
Comment (ValentinMerlet) [inline src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/FindAllMeasurementFamilies.php:29]: You don't need the variable but maybe you want to keep it for visibility.

File: src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/MeasurementFamily.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Tool\Bundle\MeasureBundle\PublicApi\Onboarder;
+
+use \Akeneo\Tool\Bundle\MeasureBundle\Model\MeasurementFamily as MeasurementFamilyAggregate;
+
+/**
+ * Read model dedicated to onboarded synchronization
```
Comment (ValentinMerlet) [inline src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/MeasurementFamily.php:10]:
```suggestion
 * Read model dedicated to onboarder synchronization
```

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+use Akeneo\Test\Integration\TestCase;
+
+/**
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class FindAllMeasurementFamiliesIntegration extends TestCase
+{
+    private FindAllMeasurementFamilies $findAllMeasurementFamilies;
```
Comment (ValentinMerlet) [inline src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php:17]: Any reason you define a property for this query? It seems that you are using it in the test `it_finds_all_measurements_families_for_onboarder` only.

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+                    ],
+                ],
+                'symbol' => 'rev',
+            ],
+        ];
+    }
+
+    /**
+     * @return string[]
+     *
```
Comment (ValentinMerlet) [inline src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php:179]: extra line

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->findAllMeasurementFamilies = $this->get(
+            'akeneo_measurement.public_api.onboarder.find_all_measurement_families'
+        );
+    }
+
+    /** @test */
+    public function it_finds_all_measurements_families_for_onboarder()
```
Comment (ValentinMerlet) [inline src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php:28]:
```suggestion
    public function it_finds_all_measurement_families_for_onboarder()
```

**Review by ValentinMerlet — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/FindAllMeasurementFamilies.php (hunk 2 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/FindAllMeasurementFamilies.php:29]: yes, exactly for visibility :)

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php (hunk 4 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php:17]: actually at first, i didn't and then i wanted to have the IDE suggestions regarding variables (i could use @var phpdoc) but i thought to myself: "what the hell maybe we'll evolve this test class with more cases in the future, let's just create a proporty for it)

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php (hunk 6 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php:28]: i'm pretty sure we say several measurement families :)

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php (hunk 5 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php:179]: thanks :)

File: src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/FindAllMeasurementFamilies.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class FindAllMeasurementFamilies
+{
+    private MeasurementFamilyRepositoryInterface $measurementFamilyRepository;
+
+    public function __construct(MeasurementFamilyRepositoryInterface  $measurementFamilyRepository)
```
Comment (ValentinMumble) [inline src/Akeneo/Tool/Bundle/MeasureBundle/PublicApi/Onboarder/FindAllMeasurementFamilies.php:19]:
```suggestion
    public function __construct(MeasurementFamilyRepositoryInterface $measurementFamilyRepository)
```

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php (hunk 6 above)
Comment (ValentinMerlet) [reply src/Akeneo/Tool/Bundle/MeasureBundle/tests/Integration/PublicApi/Onboarder/FindAllMeasurementFamiliesIntegration.php:28]: We do, but I thought that, about grammar, the `s` was only on families and not on measurement

**Pull request merged**